### DISCUSSION TITLE
How to set an empty array as a query's initial value without it inferring undefined?

### DISCUSSION THREAD

**Discussion by franciscohermida** [posva/pinia-colada#108] [Questions] [answered]
Considering this example:

``` ts
const todos = ref([]) // initial value is [] so I don't have to check for undefined in the code

async function fetchData(){
  todos.value = await apiCall() // <- always returns array
}
fetchData()
```

**Comment by posva** [comment, marked as answer]
Pass it with `initialData` or `placeholderData`

- https://pinia-colada.esm.dev/api/@pinia/colada/interfaces/UseQueryOptions.html#initialData-
- https://pinia-colada.esm.dev/api/@pinia/colada/interfaces/UseQueryOptions.html#placeholderData-

**Reply by franciscohermida** [reply]
Even if I use placeholder or initial data the type still includes undefined ``` Todo[] | undefined ```  compared to ```ref<Todo[]>([])```. 

I have to do undefined checks everywhere to satisfy TS compared to the simple approach using refs.

**Reply by posva** [reply]
Yes, because the data can still be set to undefined afterwards or placeholder data can also return undefined

**Reply by franciscohermida** [reply]
So there is no way to set the type to ```Todo[]``` instead of ```Todo[] | undefined``` in pinia colada? 

> Yes, because the data can still be set to undefined afterwards or placeholder data can also return undefined

If the type is ```Todo[]``` Typescript should complain if I assign undefined to the Ref afterwards. I make sure to always return empty array if something goes wrong. Placeholder is also empty array

**Reply by posva** [reply]
Right now you have to cast it but I should be able to improve this type in particular in the future

**Reply by posva** [reply]
Should be fixed by https://github.com/posva/pinia-colada/pull/114 :tada:

**Reply by franciscohermida** [reply]
Thank you!

A small token to show my appreciation

![image](https://github.com/user-attachments/assets/d578fdd4-799f-4d33-abae-c9d7381f4720)

**Reply by posva** [reply]
Thank you! I appreciate it 😄
